python/message-service/message_service/__main__.py

```diff
@@ -1,37 +1,38 @@
 import os
-import beeline
-import beeline.propagation.w3c as w3c
-import libhoney
 import random
-from bottle import Bottle, run
-from beeline.middleware.bottle import HoneyWSGIMiddleware
-
-beeline.init(
-    # Get this via https://ui.honeycomb.io/account after signing up for Honeycomb
-    writekey=os.environ.get("HONEYCOMB_API_KEY"),
-    # The name of your app is a good choice to start with
-    api_host=os.environ.get("HONEYCOMB_API_ENDPOINT"),
-    service_name='message-python',
-    debug=True,
-    http_trace_parser_hook=w3c.http_trace_parser_hook,
-    http_trace_propagation_hook=w3c.http_trace_propagation_hook,
-)
 
+from bottle import Bottle, run
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import \
+    OTLPSpanExporter
+from opentelemetry.instrumentation.requests import RequestsInstrumentor
+from opentelemetry.instrumentation.wsgi import OpenTelemetryMiddleware
+from opentelemetry.sdk.resources import Resource, SERVICE_NAME
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
 
 messages = [
     "how are you?", "how are you doing?", "what's good?", "what's up?", "how do you do?",
     "sup?", "good day to you", "how are things?", "howzit?", "woohoo",
 ]
-
+trace.set_tracer_provider(TracerProvider(
+    resource=Resource.create({SERVICE_NAME: "message-python"})
+))
+tracer = trace.get_tracer_provider().get_tracer(__name__)
+
+trace.get_tracer_provider().add_span_processor(
+    BatchSpanProcessor(OTLPSpanExporter(
+        headers=(("x-honeycomb-team", os.environ.get("HONEYCOMB_API_KEY")),),
+        endpoint=os.environ.get("HONEYCOMB_API_ENDPOINT",
+                                "https://api.honeycomb.io")
+    )))
 
 app = Bottle()
-
+app.wsgi = OpenTelemetryMiddleware(app.wsgi)
+RequestsInstrumentor().instrument()
 
 @app.route('/message')
 def message():
     return random.choice(messages)
```

Comment:
It might be nice as a bonus to add a manual trace that's something like "🤖 choosing message", [similar to a manual trace we have on name service](https://github.com/honeycombio/example-greeting-service/blob/main/python/name-service/name_service/__init__.py#L62).

Reply:
Done ✅